### ISSUE TITLE
HttpUnknownRequestResponse is no longer used since PR #6581

### ISSUE THREAD

**Issue by qgrosperrin** [rapid7/metasploit-framework#7049]
## Steps to reproduce

1. Get the latest version of metasploit ;)
2. Create a reverse_http handler (or any other handler derived from it, such as reverse_https). My msfconsole resource file:
```
setg PROMPT "%T - 172.16.145.182 (s:%S j:%J) msf"
spool /root/.msf-spool.log
setg AutoSystemInfo false
setg VERBOSE true
setg EnableStageEncoding true
setg ConsoleLogging true  
setg LogLevel 5  
setg SessionLogging true  
setg TimestampOutput true
setg STORE_LOOT true
setg LHOST 172.16.145.182

set SessionCommunicationTimeout 3600
set PrependMigrate true
set PrependMigrateProcess explorer.exe
set ExitOnSession false
setg MeterpreterUserAgent Mozilla/5.0 (Windows NT 6.1; Trident/7.0; rv:11.0) like Gecko
setg HttpUnknownRequestResponse <html><head><title>403 Forbidden</title></head><body bgcolor="white"><center><h1>403 Forbidden</h1></center><hr><center>nginx</center></body></html>
setg MeterpreterServerName nginx

use multi/handler
set PAYLOAD windows/meterpreter/reverse_https
set LPORT 443
exploit -j
```

3. Make sure the `HttpUnknownRequestResponse` option is specified
4. Send a request to the handler on any URL (other than for the payload itself)
5. The request is showing in the console, but no response is sent. Example output from console:
```
2016-07-01 09:52:40 +0100 - 172.16.145.182 (s:0 j:1) msf exploit(handler) > 
[*] [2016.07.01-09:52:40] Started HTTPS reverse handler on https://172.16.145.182:443
[*] [2016.07.01-09:52:40] Starting the payload handler...
2016-07-01 09:52:40 +0100 - 172.16.145.182 (s:0 j:1) msf exploit(handler) > [*] [2016.07.01-09:53:17] https://172.16.145.182:443 handling request from 172.16.145.1; (UUID: iwwbl0ga) Unknown request to /asdfd with UA 'curl/7.30.0'
```

## Expected behavior

When the `HttpUnknownRequestResponse` option is used, the framework should return the content of that option in HTTP responses.

## Current behavior

All responses to any page (other than the payload's) are empty
```
~ $ curl -k https://172.16.145.182:443/asdfd
curl: (52) Empty reply from server
```

## System stuff
kali 2, also tested on debian.

### Metasploit version
Latest git version (as of yesterday)

```
Framework: 4.12.10-dev-afbeb2b
Console  : 4.12.10-dev-afbeb2b
```

## Affected code
So I tried to look at the code to find the underlying problem, and it looks like this issue was introduced with: 3b893cf740ce8dcce6e85135308acabe3fbf238c

```ruby
         unless [:unknown_uuid, :unknown_uuid_url].include?(info[:mode])
            print_status("Unknown request to #{request_summary}")
          end
 -        resp.code    = 200
 -        resp.message = 'OK'
 -        resp.body    = datastore['HttpUnknownRequestResponse'].to_s
 +        resp = nil
          self.pending_connections -= 1
      end
  
      cli.send_response(resp) if (resp)
```

It is clear that the use case for `HttpUnknownRequestResponse` was removed and from that date, it's never called anywhere in the reverse_http.rb code.

I don't know if it's expected behaviour that any unknown request should now send an empty response, but if the `HttpUnknownRequestResponse` is used, it would probably make sense not to.

Note that obviously the same behaviour is observed whether `HttpUnknownRequestResponse` is used or not.

**Comment by bcook-r7** [comment]
Hmm, I'll take responsibility for this. The fix I made should be reverted and perhaps addressed at the stager level instead.

**Comment by bcook-r7** [comment]
Fixed in #7066, thanks for the note (and sorry for the regression!)
